dj/sql/parsing/backends/sqloxide.py

```diff
@@ -283,13 +291,16 @@ def parse_function(parse_tree: dict) -> Function:
     if match_keys_subset(parse_tree, {"name", "args"}):
         args = parse_tree["args"]
         names = parse_tree["name"]
-        if len(names) != 1:
-            raise DJParseException("Expected a single name for Function")
+
         return cast(
             Function,
             Function(
-                names[0]["value"],
-                names[0]["quote_style"] if names[0]["quote_style"] is not None else "",
+                cast(
```

Comment:
Just looked up `cast` and learned something new. Does this boil down to an alternative to including `# type: ignore` in other places? What made you use `cast` instead?

Reply:
First I saw it in @betodealmeida's code. To me it seems like bad practice to handle it with the ignore if it can be otherwise handled with the cast. Otherwise, I wonder if the effects could cascade to where ignoring would still imply that the value is just `Node` and not be specific (in this case `Function`) - I sort of doubt this since all the attributes are still typed but at least this also clarifies the intend ad recognition of the actual problem which is that the base class `Node` defines the method and only specifies the `Node` type as the return type. I tried using the `Self` type but mypy complained that that could not be used as a type. Reading up on `Self` https://peps.python.org/pep-0673/ suggested it should be able to be used like that but apparently not yet or at least not with the versions we allow (I imported from typing-extensions)

Reply:
Yeah, `Self` is recent: https://peps.python.org/pep-0673/